### PR TITLE
feat(ui): Add new properties to `UnableToDecryptInfo`

### PR DESCRIPTION
**Pull request by richvdh**
Adds a number of new properties to the `UnableToDecryptInfo` that is passed to the application-level `UnableToDecryptHook`.

Part of https://github.com/element-hq/element-meta/issues/2582

### CODE REVIEW

File: crates/matrix-sdk-ui/src/unable_to_decrypt_hook.rs [hunk 1]
```diff
@@ -63,6 +63,21 @@ pub struct UnableToDecryptInfo {
...
+
+    /// Whether the user had verified their own identity at the point they
+    /// received the UTD event.
+    pub user_trusts_own_identity: bool,
+
+    /// True if the sender of the event was on a different server to our user.
+    pub is_federated: bool,
+
+    /// True if our user is using the `matrix.org` homeserver.
+    pub is_matrix_dot_org: bool,
```
Comment (BillCarsonFr) [inline crates/matrix-sdk-ui/src/unable_to_decrypt_hook.rs:80]: It feels strange to have a `is_matrix_dot_org` at the `sdk` level. 
Can we have instead a sender_homerserver and a own_homeserver , then the app could do a  `equals("matrix.org")`  and the is_federated?

Comment (richvdh) [reply crates/matrix-sdk-ui/src/unable_to_decrypt_hook.rs:80]: I was torn on this. On the one hand, I agree: this does seem a bit arbitrary. On the other, I was keen to factor out a as much as possible of the EXA/EXI common code.

Comment (Hywan) [reply crates/matrix-sdk-ui/src/unable_to_decrypt_hook.rs:80]: I think @BillCarsonFr's idea is quite nice. I understand your position, but I think it's important to not take this kind of shortcuts/biases in the SDK.

**Review by Hywan — CHANGES_REQUESTED** [review]
Seems good! One note about `is_matrix_dot_org` that I would prefer to not see here. Otherwise, great patches!

File: crates/matrix-sdk-ui/src/unable_to_decrypt_hook.rs (hunk 1 above)
Comment (richvdh) [reply crates/matrix-sdk-ui/src/unable_to_decrypt_hook.rs]: Fair enough. PTAL.

File: crates/matrix-sdk-ui/src/unable_to_decrypt_hook.rs [hunk 2]
```diff
@@ -73,11 +73,12 @@ pub struct UnableToDecryptInfo {
     /// received the UTD event.
     pub user_trusts_own_identity: bool,
 
-    /// True if the sender of the event was on a different server to our user.
-    pub is_federated: bool,
+    /// The homserver of the user that sent the undecryptable event.
```
Comment (Hywan) [inline crates/matrix-sdk-ui/src/unable_to_decrypt_hook.rs:76]:
```diff
- homserver
+ homeserver
```

**Review by Hywan — APPROVED** [review]
Thanks for the fixup! A last typo in the doc, but I'm approving the PR to avoid a new review cycle.

File: crates/matrix-sdk-ui/src/unable_to_decrypt_hook.rs (hunk 2 above)
Comment (richvdh) [reply crates/matrix-sdk-ui/src/unable_to_decrypt_hook.rs:76]:
```suggestion
    /// The homeserver of the user that sent the undecryptable event.
```

**Pull request merged**